pkg/network/master/subnets.go

```diff
@@ -55,15 +55,68 @@ func (master *OsdnMaster) SubnetStartMaster(clusterNetworks []common.ClusterNetw
 	}
 	master.subnetAllocatorList = subnetAllocatorList
 
+	err = master.verifyHostSubnetsState(subnets.Items)
+	if err != nil {
+		return err
+	}
+
 	master.watchNodes()
 	master.watchSubnets()
 	return nil
 }
 
+func (master *OsdnMaster) verifyHostSubnetsState(hostSubnets []networkapi.HostSubnet) error {
+	nodes, err := master.kClient.Core().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Error in initializing/fetching nodes: %v", err))
+		return err
+	}
+	nodeUIDmap := make(map[string]string)
+	for _, node := range nodes.Items {
+		nodeUIDmap[node.Name] = string(node.UID)
+	}
+	for _, subnet := range hostSubnets {
```

Comment:
subnet.Annotations could be nil and subnet.Annotations["nodeUID"] can fail in subsequent calls.
You can initialize this variable before performing any checks:
```
if subnet.Annotations == nil {
   subnet.Annotations = make(map[string]string)
}
```

Reply:
a nil map is fine to query for values, a map will always return the "zero" value (in this case empty string) when a key does not exist.